core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala

```diff
@@ -6,40 +6,22 @@ import com.databricks.labs.remorph.parsers.snowflake.SnowflakeParser._
 import scala.collection.JavaConverters._
 class SnowflakeExpressionBuilder extends SnowflakeParserBaseVisitor[ir.Expression] {
 
-  override def visitSelect_list_elem(ctx: SnowflakeParser.Select_list_elemContext): ir.Expression = {
-    if (ctx.column_elem() != null) {
-      val column = ctx.column_elem().accept(this)
-      if (ctx.as_alias() != null) {
-        ctx.as_alias().accept(this) match {
-          case ir.Alias(_, name, metadata) => ir.Alias(column, name, metadata)
-          case _ => null
-        }
-      } else {
-        column
-      }
-    } else if (ctx.expression_elem() != null) {
-      ctx.expression_elem().accept(this)
-    } else {
-      null
+  override def visitSelect_list_elem(ctx: Select_list_elemContext): ir.Expression = {
+    val rawExpression = ctx match {
+      case c if c.column_elem() != null => c.column_elem().accept(this)
+      case c if c.expression_elem() != null => c.expression_elem().accept(this)
+      case _ => null
```

Comment:
Bug: "input: ir.Expression" becomes null, which it shouldn't. I guess you have to use Option here

Reply:
It's actually a not-yet-implemented case, let me just implement it properly. Stay tuned